core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java

```diff
@@ -194,8 +203,24 @@ TableOperations newTableOps(String location) {
     if (location.contains(METADATA_JSON)) {
       return new StaticTableOperations(location, new HadoopFileIO(conf));
     } else {
-      return new HadoopTableOperations(new Path(location), new HadoopFileIO(conf), conf);
+      return new HadoopTableOperations(new Path(location), new HadoopFileIO(conf), conf,
+              createOrGetLockManager(location));
```

Comment:
This class should only need one `LockManager`. The same `LockManager` is thread safe and can be shared by all the `TableOperations`. That's how the original use in `GlueCatalog` works.

Reply:
hmmm...the consideration here is actually users can pass different configurations of LockManager for different tables

Reply:
The `LockManager` is configured by the Hadoop `Configuration`, which is passed into `HadoopTables` at create time. While this does implement `Configurable`, I don't think there is any need to support changing lock manager configurations.

If we tried to support configuration changes we would introduce inconsistency: what happens when a table is loaded, the config changes, and then the table is loaded again? Either you use the same lock manager with the old settings, or you create a different lock manager and break in-memory locks. There is no need to introduce these worries.

To avoid unnecessary complexity, I think we should remove the cache and instantiate a lock manager in the constructor that is used for all tables loaded by this interface. Then we don't have to worry about having a lock manager per table -- that's a good thing because each lock manager has a heartbeat thread!